metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -3168,11 +3214,63 @@ def emit_site_container(self):
             sub_types=[c.SITE],
         )
 
+    def _fetch_groups(self):
+        for group in TSC.Pager(self.server.groups):
+            self.group_map[group.id] = group
+
+    def _filter_capabilities(self, capabilities):
+        if not self.config.group_ingestion:
+            return ""
+
+        allowed_capabilities = [
+            key for key, value in capabilities.items() if value == "Allow"
+        ]
+        if self.config.group_ingestion.displayed_capabilities:
+            allowed_capabilities = [
+                value
+                for value in allowed_capabilities
+                if value in self.config.group_ingestion.displayed_capabilities
+            ]
+        return ", ".join(allowed_capabilities)
+
+    def _create_workbook_properties(
+        self, permissions: List[PermissionsRule]
+    ) -> Optional[Dict[str, str]]:
+        if not self.config.group_ingestion:
+            return None
+
+        groups = []
+        for rule in permissions:
+            if rule.grantee.tag_name == "group":
+                group = self.group_map.get(rule.grantee.id)
+                if not group or not group.name:
+                    logger.debug(f"Group {rule.grantee.id} not found in group map.")
+                    continue
+                if not self.config.group_ingestion.group_name_pattern.allowed(
+                    group.name
+                ):
+                    logger.info(
+                        f"Skip permission '{group.name}' as it's excluded in group_name_pattern."
+                    )
+                    continue
+
+                capabilities = (
+                    f" ({self._filter_capabilities(rule.capabilities)})"
+                    if self._filter_capabilities(rule.capabilities)
+                    else ""
+                )
+                groups.append(f"{group.name}{capabilities}")
+
+        return {"groups": ", ".join(groups)} if len(groups) > 0 else None
```

Comment:
might be easier to just `json.dumps(groups)`

Reply:
So would you rather want to see the groups with their capabilities as a JSON string in the UI? Would be easier to parse later in the transformer but could be a bit hard on the eyes in the custom properties.

Reply:
It now looks like this: ![image](https://github.com/user-attachments/assets/d36febc2-1a81-4368-8878-3c750f901ab4)